### PR TITLE
Feature/autogenstudio UI extra headers

### PR DESCRIPTION
**Pull request by ronaldpereira**
## Why are these changes needed?

**Problem:** The autogenstudio library lacked the ability to specify default headers for API requests. This caused issues when APIs required specific headers to be included in every request.

**Solution:** I've extended autogenstudio to accept an optional default_headers parameter. This parameter allows developers to define a dictionary containing the desired headers, which will be automatically included in all API requests generated by autogenstudio.

For example, when any custom API needed some extra headers to complete the request:
`"default_headers": {"X-Requester-Token": "<requester_token>"}`
The llm_config parameter already accepts this configuration, but it wasn't possible to pass it via the autogenstudio via model creation.

## Checks

- [x] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

File: autogen/oai/client.py [hunk 1]
```diff
@@ -411,6 +412,10 @@ def _register_default_client(self, config: Dict[str, Any], openai_config: Dict[s
         """
         openai_config = {**openai_config, **{k: v for k, v in config.items() if k in self.openai_kwargs}}
         api_type = config.get("api_type")
+        default_headers = config.get("default_headers")
+        if isinstance(default_headers, str):
+            openai_config["default_headers"] = json.loads(default_headers)
```
Comment (sonichi) [inline autogen/oai/client.py:417]: How about documenting this behavior in docstr, and making a test to cover this use case?

Comment (ronaldpereira) [reply autogen/oai/client.py:425]: Thank you for your review. Both suggestions were done!

File: test/oai/test_client.py [hunk 2]
```diff
@@ -299,12 +299,56 @@ def test_cache():
...
+        "default_headers": {"Content-Type": "application/json"},
+    }
+    openai_config = {"base_url": "https://api.openai.com", "api_key": "YOUR_API_KEY"}
+    client = OpenAIWrapper(config_list=[config])
+    client._register_default_client(config, openai_config)
+    assert isinstance(client._clients[0], PlaceHolderClient)
+    assert client._clients[0].config == config
+
+
+@pytest.mark.skipif(skip, reason="openai>=1 not installed")
```
Comment (sonichi) [inline test/oai/test_client.py:316]: same

File: test/oai/test_client.py [hunk 3]
```diff
@@ -299,12 +299,56 @@ def test_cache():
...
+    client._register_default_client(config, openai_config)
+    assert client._clients[0].config.default_headers == {"Another-Header": "HEADER_TEST"}
+    assert isinstance(client._clients[0], OpenAIClient)
+    assert isinstance(client._clients[0]._oai_client, AzureOpenAI)
+    assert client._clients[0]._oai_client.base_url == "https://api.openai.com/openai/"
+    assert client._clients[0]._oai_client.api_key == "YOUR_API_KEY"
+    assert client._clients[0]._oai_client.default_headers.get("Another-Header") == "HEADER_TEST"
+
+
+@pytest.mark.skipif(skip, reason="openai>=1 not installed")
```
Comment (sonichi) [inline test/oai/test_client.py:330]: same

**Review by sonichi — COMMENTED** [review]
Since this PR requires openai test, could you please create a branch in the upstream repo and make a PR from that branch? Thanks.

**Comment by ronaldpereira** [comment]
> Since this PR requires openai test, could you please create a branch in the upstream repo and make a PR from that branch? Thanks.

Done: [New Upstream MR](https://github.com/microsoft/autogen/pull/2433)

**Pull request closed without merging**